connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ConnectProgressExecutionListener.scala

```diff
@@ -82,10 +82,14 @@ private[connect] class ConnectProgressExecutionListener extends SparkListener wi
      *
      * If the tracker was marked as dirty, the state is reset after.
      */
-    def yieldWhenDirty(thunk: (Seq[StageInfo], Long) => Unit): Unit = {
-      if (dirty.get()) {
+    def yieldWhenDirty(force: Boolean = false, thunk: (Seq[StageInfo], Long) => Unit): Unit = {
```

Comment:
you may want to defined it as 
```suggestion
    def yieldWhenDirty(force: Boolean = false)(thunk: (Seq[StageInfo], Long) => Unit): Unit = {
```
then the caller side could be written in
```
tracker.yieldWhenDirty(force) { (stages, inflightTasks) =>
  ...
}
```

Reply:
Yes correct, i was looking for this particular syntax. Will change to that. Thanks.